src/kudu/consensus/consensus_queue.cc

```diff
@@ -879,15 +879,18 @@ void PeerMessageQueue::AdvanceQueueWatermark(const char* type,
 }
 
 void PeerMessageQueue::BeginWatchForSuccessor(
-    const boost::optional<string>& successor_uuid) {
+    const boost::optional<string>& successor_uuid,
+    const std::function<bool(const kudu::consensus::RaftPeerPB&)>& filter_fn) {
```

Comment:
Would we pass a functor along or should we deal with just UUIDs at this layer?

Reply:
i don't fully understand this comment. We have to pass functor along, because the selection is async.
we might have multiple UUIDs which are not master ban filtered, and you would like to select the first one that is fully caught up.